compiler/cpp/src/generate/t_cpp_generator.cc

```diff
@@ -1626,6 +1640,16 @@ void t_cpp_generator::generate_struct_ostream_operator_decl(std::ofstream& out,
   out << "std::ostream& operator<<(std::ostream& out, const " << tstruct->get_name() << "& obj)";
 }
 
+void t_cpp_generator::generate_exception_what_method_decl(std::ofstream& out,
+                                                          t_struct* tstruct,
+                                                          bool external) {
+  std::string prefix = "";
+  if (external) {
+    prefix = string(tstruct->get_name() + "::");
+  }
+  out << "const char* " << prefix << "what() const throw()";
```

Comment:
It may be, but TException implements what() using this definition. Removing "throw()" results in compilation error: "overriding 'virtual const char* apache::thrift::TException::what() const throw ()'".

However, I believe I can just implement it without concatenating strings specifically.